fdbclient/ParallelStream.actor.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * ParallelStream.actor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+// When actually compiled (NO_INTELLISENSE), include the generated version of this file.  In intellisense use the source
+// version.
+#if defined(NO_INTELLISENSE) && !defined(FDBCLIENT_PARALLEL_STREAM_ACTOR_G_H)
+#define FDBCLIENT_PARALLEL_STREAM_ACTOR_G_H
+#include "fdbclient/ParallelStream.actor.g.h"
+#elif !defined(FDBCLIENT_PARALLEL_STREAM_ACTOR_H)
+#define FDBCLIENT_PARALLEL_STREAM_ACTOR_H
+
+#include <deque>
+#include <vector>
+
+#include "flow/genericactors.actor.h"
+#include "flow/actorcompiler.h" // must be last include
+
+template <class T>
+class ParallelStream {
```

Comment:
The stream should be reference counted, and the fragments should hold a reference to the stream, so that way we do not destroy the stream until all fragments have been destroyed